### ISSUE TITLE
Android real device

### ISSUE THREAD

**Issue by jmc420** [sbugert/react-native-admob#158]
Hi

I have simple example working fine with both the iOS and Simulators but I am struggling to get it working on an Android device. I get ERROR_CODE_INTERNAL_ERROR back in the didFailToReceiveAdWithError callback.

This has been discussed here:

https://github.com/sbugert/react-native-admob/issues/76

Some people there say you need google-services.json? Is this the problem?

James

**Comment by jmc420** [comment]
I added google-services.json and it made no difference.

I tried it on an ipad and got this error: "Request Error: No ad to show"

Is there something you need to do to make ads work on real devices?

**Comment by jmc420** [comment]
I tried linking to Firebase and added google-services.json but this made no difference.

I have posted my example code on github:

https://github.com/jmc420/react-native-examples/tree/master/AdmobExample

Initially this worked fine on both iOS and Android simulator but not on devices. 

On an Android device I got error ERROR_CODE_INTERNAL_ERROR (0) returned from Google admob. On an Apple device I got a more useful error message:

2017-07-26 15:56:53.301 AdmobExample[216:7012] <Google> To get test ads on this device, call: request.testDevices = @[ @"caec8d3e28473f38a97b7905b1f9fc81" ];

I then followed https://developers.google.com/admob/ios/test-ads#enable_test_devices and the app started working on Android but I gave up trying to getting it to work on an Apple device.

Hope this helps someone!
